db/functions/packed.py

```diff
@@ -70,6 +70,124 @@ def unpack(self):
         ])
 
 
+class ArrayLengthEquals(DBFunctionPacked):
+    id = 'array_length_equals'
+    name = 'Number of elements is'
+    hints = tuple([
+        hints.returns(hints.boolean),
+        hints.parameter_count(2),
+        hints.parameter(0, hints.array),
+        hints.parameter(1, hints.string_like),
```

Comment:
The parameter against which you'll be comparing array lengths should be an integer (unless I'm missing something).

Reply:
That's exactly it. With hints we try to describe various aspects of DBFunctions, for the benefit of frontend. There are no hard rules here, more like hard conventions.

Something to note is that the hint system isn't being used atm, because at some point frontend had a problem (I don't recall what it was) and we stubbed the relevant things on the frontend, and postponed fixing that underlying problem. Hopefully it's something we can get to after release.